server/memphis_handler_schemas.go

```diff
@@ -194,21 +194,34 @@ func (sh SchemasHandler) CreateNewSchema(c *gin.Context) {
 		Versions: []primitive.ObjectID{newSchemaVersion.ID},
 	}
 
-	_, err = schemasCollection.InsertOne(context.TODO(), newSchema)
+	filter := bson.M{"name": newSchema.Name}
+	update := bson.M{
+		"$setOnInsert": bson.M{
+			"_id":  newSchema.ID,
+			"type": newSchema.Type,
+		},
+		"$set": bson.M{"versions": newSchema.Versions},
```

Comment:
This is also on insert only, otherwise it will getting updated more than once in a case of 2 parallel requests